source/agora/consensus/EnrollmentManager.d

```diff
@@ -33,9 +33,47 @@ import d2sqlite3.sqlite3;
 import std.algorithm;
 import std.file;
 import std.path;
+import std.range;
 
 mixin AddLogger!();
 
+/*******************************************************************************
+    Store preimages of seed data
+*******************************************************************************/
+
+public struct PreimageStore
+{
+    /// Preimages of hashes of random value
+    public Hash[] data;
```

Comment:
We could avoid writing all this code if we added serialization support to `Hash[]` in `Serializer.d` and `Deserializer.d`.

@Geod24 can you think of any drawbacks?

Reply:
After adding serialization support to `Hash[]` in this PR, I will request a review again.

Reply:
@AndrejMitrovic : Actually for the binary protocol, I wrote a way to serialize ranges... Guess it's time to PR it.